packages/dev/docs/src/docs.js

```diff
@@ -31,7 +31,7 @@ for (let link of links) {
     <ThemeProvider UNSAFE_className={docsStyle.inlineProvider}>
       <DialogTrigger type="popover">
         <Pressable>
-          <a href={link.href} data-link={link.dataset.link} className={link.className} onClick={e => e.preventDefault()}>{link.textContent}</a>
+          <a href={link.href} data-link={link.dataset.link} className={link.className} aria-haspopup="dialog" onClick={e => e.preventDefault()}>{link.textContent}</a>
```

Comment:
I think the problem is that NVDA and JAWS announce all buttons with with `aria-haspopup` as "menu" or a "subMenu" button, which implies a keyboard navigation behavior that a dialog doesn't have, and the core-aam spec isn't specific enough about what the expected behavior should be: https://github.com/w3c/core-aam/issues/51.

VoiceOver behaves as expected: https://a11ysupport.io/tests/tech__aria__aria_haspopup_attribute

There also seems to be an issue with the presumption of it being a menu causing the screen reader to switch into forms mode, which is problematic for a dialog, where you may need to read content, because pressing Esc to exit forms mode may also close the dialog:

https://github.com/w3c/aria/issues/1024#issuecomment-574844836

I can remove the `aria-haspopup="dialog"`, especially after removing the CSS that add the down arrow as generated content.